clients/src/main/java/org/apache/kafka/clients/consumer/NoOffsetForPartitionException.java

```diff
@@ -26,19 +26,20 @@
 
     private static final long serialVersionUID = 1L;
 
-    private final TopicPartition partition;
+    private final Set<TopicPartition> partitionSet;
 
-    public NoOffsetForPartitionException(TopicPartition partition) {
-        super("Undefined offset with no reset policy for partition: " + partition);
-        this.partition = partition;
+    public NoOffsetForPartitionException(String message) {
```

Comment:
@ewencp This exception may not need this constructor, but some of the other exceptions in this patch which have some contextual error state need similar constructors for the static exception mapping in Errors. My point above was mainly that the usage of these exceptions should be consistent. We either support this internal state (in which case it should be filled properly) or we don't. I had intentionally avoided adding the mapping in Errors before because of this reason, preferring instead to leave them as generic ApiException instances.